Sources/MessagingPush/PushHandling/iOSPushEventListener.swift

```diff
@@ -0,0 +1,138 @@
+            // push did not come from CIO
+            // Do not call completionHandler() because push did not come from CIO.
+            // Forward the request to all other push click handlers in app to give them a chance to handle it.
+            notificationCenterDelegateProxy.userNotificationCenter(center, willPresent: notification, withCompletionHandler: completionHandler)
+
+            return
+        }
+
+        // Push came from CIO. Handle the push on behalf of the customer.
+        // Make sure to call completionHandler() so the customer does not need to.
+
+        if moduleConfig.showPushAppInForeground {
+            // Tell the OS to show the push while app in foreground using 1 of the below options, depending on version of OS
+            if #available(iOS 14.0, *) {
+                completionHandler([.list, .banner, .badge, .sound])
+            } else {
+                completionHandler([.badge, .sound])
+            }
+        } else {
+            completionHandler([]) // do not show push while app in foreground
+        }
+    }
+}
+
+// This is a bit confusing and makes the code a little more complex. However, it's the most reliable way found to get UNUserNotificationCenter.delegate swizzling to work by using an extension.
+@available(iOSApplicationExtension, unavailable)
+extension UNUserNotificationCenter {
+    // Swizzled method that gets called when `UNUserNotificationCenter.current().delegate` setter called.
+    @objc dynamic func cio_swizzled_setDelegate(delegate: UNUserNotificationCenterDelegate?) {
+        guard let diGraph = SdkInitializedUtilImpl().postInitializedData?.diGraph else {
+            // SDK not initialized yet. We hope that this function never gets called because auto push click handling will not work if this function called.
+
+            // Forward request to the original implementation implementation that we swizzled. So that the app finishes setting UNUserNotificationCenter.delegate.
+            // We have to provide the original delegate object because the SDK has not yet been initialized.
+            cio_swizzled_setDelegate(delegate: delegate)
+
+            return
+        }
+
+        diGraph.logger.debug("New UNUserNotificationCenter.delegate set. Delegate class: \(String(describing: delegate))")
+
+        diGraph.pushEventListener.newNotificationCenterDelegateSet(delegate)
+
+        // Forward request to the original implementation that we swizzled. So that the app finishes setting UNUserNotificationCenter.delegate.
+        //
+        // Instead of providing the given 'delegate', provide CIO SDK's click handler.
+        // This will force our SDK to be the 1 push click handler of the app instead of the given 'delegate'.
+        cio_swizzled_setDelegate(delegate: diGraph.pushEventListener.delegate)
+    }
+}
+
+// TODO: cleanup duplicate swizzle functions all over codebase.
```

Comment:
Is this still pending?

Reply:
thanks for asking. 

This is taken care of in the lint fix PR.